bugsnag-plugin-android-anr/src/main/java/com/bugsnag/android/AnrPlugin.kt

```diff
@@ -5,26 +5,30 @@ import android.os.Looper
 
 internal class AnrPlugin : Plugin {
 
-    companion object {
-        init {
-            System.loadLibrary("bugsnag-plugin-android-anr")
-        }
-    }
-
+    private val loader = LibraryLoader()
     private lateinit var client: Client
     private val collector = AnrDetailsCollector()
 
     private external fun enableAnrReporting(callPreviousSigquitHandler: Boolean)
     private external fun disableAnrReporting()
 
     override fun load(client: Client) {
-        // this must be run from the main thread as the SIGQUIT is sent to the main thread,
-        // and if the handler is installed on a background thread instead we receive no signal
-        Handler(Looper.getMainLooper()).post(Runnable {
-            this.client = client
-            enableAnrReporting(true)
-            client.logger.w("Initialised ANR Plugin")
-        })
+        val loaded = loader.loadLibrary("bugsnag-plugin-android-anr", client) {
+            val error = it.errors[0]
+            error.errorClass = "AnrLinkError"
+            error.errorMessage = "Native library could not be linked. Bugsnag will not report ANRs."
+            true
+        }
+
+        if (loaded) {
+            // this must be run from the main thread as the SIGQUIT is sent to the main thread,
+            // and if the handler is installed on a background thread instead we receive no signal
+            Handler(Looper.getMainLooper()).post(Runnable {
+                this.client = client
+                enableAnrReporting(true)
+                client.logger.w("Initialised ANR Plugin")
+            })
+        }
```

Comment:
Shall we log a warning to the `logger` if it fails to load as well? Alternatively perform the log message in LibraryLoader and log "ANR reporting enabled" above.